woocommerce/woocommerce-ios
# Product Settings: implemented the changing status of a Product

Resolve #1823 

## Description
This functionality is under the feature flag M2.
This PR implements the changing status of a Product inside Product Settings, it also manages the transfer of data between the various view controllers.

## Changes
- Now the `statusKey` is sent in the Product encoding
- Implemented the new `ProductStatusSettingListSelectorDataSource` + tests
- Implemented `ProductSettings` with all the editable parameters in Product Settings (for the moment only the `status`). I’m not sure about the position, now it’s declared inside `ProductSettingsViewModel`, but it could be in a separate file.
- Now `ProductSettingsSections` and `ProductSettingsRows` accept `ProductSettings`.
- New *Status* cell in Product Settings
- Now `ProductSettingsViewController` is able to reload the UI when the view model change, and it’s able to send data to the `ProductFormViewController`. Managed also the discard changes action sheet.
- `ProductSettingsViewModel` now use `ProductSettings`, implement the `onReload` closure, handle the cell tap, and has a method to check for unsaved changes.
- `ProductUpdated` now has a new method to manage the update of product settings.

## Testing
1) Navigate to a product detail
2) Press the "more" button in the navigation bar
3) Press "Product Settings" -> You are able to see the cell "Status" with the current status.
4) Press the "Status" cell, you will be able to see the 4 statuses: `publish`, `draft`, `pending`, `private`.
5) Try to change the status, press back.
6) Press "done", and update the product. -> You will be able to see the new product status in the Products List and in Product Settings.
7) Try again to edit the product status, press back, and discard changes. -> Make sure that everything is like before.

## Screenshots
| New Status cell + Done button in Product Settings             |  Status list |
:-------------------------:|:-------------------------:
![Simulator Screen Shot - iPhone 11 Pro - 2020-03-27 at 18 56 44](https://user-images.githubusercontent.com/495617/77785627-c1444380-705c-11ea-81db-5862f91d0349.png) | ![Simulator Screen Shot - iPhone 11 Pro - 2020-03-27 at 18 56 42](https://user-images.githubusercontent.com/495617/77785644-c608f780-705c-11ea-8f0b-4955c3e021a0.png)



Update release notes:
